### PR TITLE
correctly translate parameter mapping to amici plist

### PR DESCRIPTION
**Pull request by FFroehlich**
probably not properly implemented yet, waiting for petab fixes for easier debugging/testing

### CODE REVIEW

**Comment by dweindl** [comment]
> waiting for petab fixes for easier debugging/testing

Do you mean waiting for upgrade to most recent PEtab, or is something broken there?

**Comment by FFroehlich** [comment]
> > waiting for petab fixes for easier debugging/testing
> 
> Do you mean waiting for upgrade to most recent PEtab, or is something broken there?

It's all good now ;)

File: pypesto/objective/amici_objective.py [hunk 1]
```diff
@@ -569,6 +525,44 @@ def map_par_opt_to_par_sim(mapping_par_opt_to_par_sim, par_opt_ids, x):
     return par_sim_vals
 
 
+def create_plistfrom_par_opt_to_par_sim(mapping_par_opt_to_par_sim,
+                                        par_opt_ids):
+    """
+    From the optimization vector `x`, create the simulation vector according
```
Comment (yannikschaelte) [inline pypesto/objective/amici_objective.py]:
```suggestion
    From the optimization parameter ids `par_opt_ids`, create the simulation vector according
```
?

File: pypesto/objective/amici_objective.py [hunk 2]
```diff
@@ -569,6 +525,44 @@ def map_par_opt_to_par_sim(mapping_par_opt_to_par_sim, par_opt_ids, x):
...
+        optimization parameter ids.
+    par_opt_ids: array-like of str
+        The optimization parameter ids. This vector is needed to know the
+        order of the entries in x.
+
+    Returns
+    -------
+
+    plist: array-like of float
+        List of parameter indices for which sensitivity needs to be computed
```
Comment (yannikschaelte) [inline pypesto/objective/amici_objective.py]:
```suggestion
        List of parameter indices for which the sensitivity needs to be computed
```

File: pypesto/objective/amici_objective.py [hunk 3]
```diff
@@ -569,6 +525,44 @@ def map_par_opt_to_par_sim(mapping_par_opt_to_par_sim, par_opt_ids, x):
...
+    """
+    plist = []
+
+    # number of simulation parameters
+    n_par_sim = len(mapping_par_opt_to_par_sim)
+
+    # iterate over simulation parameter indices
+    for j_par_sim in range(n_par_sim):
```
Comment (yannikschaelte) [inline pypesto/objective/amici_objective.py]:
```suggestion
    for j_par_sim, val in enumerate(mapping_par_opt_to_par_sim):
```
... and remove what's before (playing codacy :) )

File: pypesto/objective/amici_objective.py [hunk 4]
```diff
@@ -622,13 +616,15 @@ def add_sim_grad_to_opt_grad(par_opt_ids,
         Coefficient for sim_grad when adding to opt_grad.
     """
 
-    for par_sim_idx, par_opt_id in enumerate(mapping_par_opt_to_par_sim):
+    par_sim_idx = 0
+    for par_opt_id in mapping_par_opt_to_par_sim:
```
Comment (yannikschaelte) [inline pypesto/objective/amici_objective.py]: could you add a comment here that we need to leave non-string overrides out of the index, as sim_grad contains only values for str overrides?
will help maintain the code later.

File: pypesto/objective/amici_objective.py [hunk 5]
```diff
@@ -646,22 +642,25 @@ def add_sim_hess_to_opt_hess(par_opt_ids,
...
         if not isinstance(par_opt_id, str):
             # this was a numeric override for which we ignore the hessian
             continue
 
         par_opt_idx = par_opt_ids.index(par_opt_id)
 
-        for par_sim_idx_2, par_opt_id_2 in enumerate(
-                mapping_par_opt_to_par_sim):
+        par_sim_idx_2 = 0
+        for par_opt_id_2 in mapping_par_opt_to_par_sim:
```
Comment (yannikschaelte) [inline pypesto/objective/amici_objective.py]: here maybe the same as in above comment, or just refer there.

**Pull request merged**